.github/workflows/clang-format.yml

```diff
@@ -0,0 +1,20 @@
+name: Style Check
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  clang-format:
+    name: Clang-Format
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: DoozyX/clang-format-lint-action@v0.12
```

Comment:
I'm rather new to Github Actions, but I think we should pin this third-party action to a specific sha1 hash. Did you review the code by this action, if not, could you do that?

(I'm aware that I'm asking for things which are not covered by existing actions either. I think it is important nevertheless and fixing it for existing workflows is on the agenda #1737).

Reply:
I had a look at the project (https://github.com/DoozyX/clang-format-lint-action) and that's fine in general.
Only downside are the binaries that are included here, so you basically have to trust them: https://github.com/DoozyX/clang-format-lint-action/tree/master/clang-format

Reply:
> I think we should pin this third-party action to a specific sha1 hash

Had a closer look at the documentation as well, and to be sure we need to use the hash, as the tag could be reused.
Changed the action accordingly.

Reply:
> Only downside are the binaries that are included here, so you basically have to trust them: https://github.com/DoozyX/clang-format-lint-action/tree/master/clang-format

That's really not optimal, but as this action should work with read-only permission only, it'd be OK, I think. The repo is public anyway, so even if there was harmful code, it would not gain any additional permissions by running there.